index.spec.ts

```diff
@@ -28,3 +27,65 @@ describe("testState", () => {
     expect(state.a).toEqual(1);
   });
 });
+
+describe("resetState", () => {
```

Comment:
Can I keep nested objects in `state`? Can I keep function references there?

Reply:
I've prepared a playground you can pull, install and test locally. 

Here is the link to repo: https://github.com/piotrszyma/teststate-playground do you have access to it?

Reply:
https://github.com/piotrszyma/teststate/pull/7 Added anti-regression tests, but it worked out of the box 👍🏻